File: src/game/tools/rpc/output/ClientCmdsInvoker.cs
using System;
using Common.Log;
using Common.Net.Simple;

namespace Game
{
	public interface IClientCmds 
	{
		void HeartBeat(Command cmd, string msg); // 心跳
		void LoginSuccess(Command cmd, PlayerWrap player, bool reconnect, string extension); // 登录成功
		void LoginFailed(Command cmd, short errorCode, string errMsg); // 登录失败（1 用户不存在，2  密码错误， 3 禁止登陆）
		
	}
	
    public class ClientCmdsInvoker : IInvoker
    {
        IClientCmds _cmds = null;
		Action<short> _onCmdInvoked = null;
        public ClientCmdsInvoker(IClientCmds cmds)
        {
            _cmds = cmds;
        }
		public void SetOnCmdInvoked(Action<short> onCmdInvoked)
		{
			_onCmdInvoked = onCmdInvoked;
		}

        public void Invoke(Command cmd)
        {
        	try
        	{
        		Packet pack = cmd.Pack;
	        	switch (cmd.Opcode)
	            {
	            	case (short)0: _cmds.HeartBeat(cmd, pack.GetString()); break;
	            	case (short)1: _cmds.LoginSuccess(cmd, new PlayerWrap().Decode(), pack.GetBool(), pack.GetString()); break;
	            	case (short)2: _cmds.LoginFailed(cmd, pack.GetShort(), pack.GetString()); break;
	            	
	            }
				if (_onCmdInvoked != null)
					_onCmdInvoked(cmd.Opcode);
        	}
        	catch(Exception e)
        	{
				L.Error("invoke error, opcode=" + cmd.Opcode);
        		L.Exception(e.Message, e);
        	}
		}
    }
}

File: src/game/tools/rpc/output/ClientCmdsInvoker.ts
interface IClientCmds
{
	HeartBeat(cmd:Net.Simple.Command, msg:string ); // 心跳
	LoginSuccess(cmd:Net.Simple.Command, player:PlayerWrap , reconnect:boolean , extension:string ); // 登录成功
	LoginFailed(cmd:Net.Simple.Command, errorCode:number , errMsg:string ); // 登录失败（1 用户不存在，2  密码错误， 3 禁止登陆）
	
}
class ClientCmdsInvoker implements Net.Simple.IInvoker {
	private _cmds:IClientCmds = null;
	private _onCmdInvoked:Function = null;
	private _obj:any = null;
	public constructor(cmds:IClientCmds) {
		this._cmds = cmds;
	}
	public SetOnCmdInvoked(func:Function,obj:any){
		this._onCmdInvoked = func;
		this._obj = obj;
	}
	public Invoke(cmd:Net.Simple.Command):void{
		var pack = cmd.Pack;
		switch(cmd.Opcode){
			case 0: _cmds.HeartBeat(cmd, new pack.GetString()); break;
			case 1: _cmds.LoginSuccess(cmd, new PlayerWrap().Decode(pack), new pack.GetBool(), new pack.GetString()); break;
			case 2: _cmds.LoginFailed(cmd, new pack.GetShort(), new pack.GetString()); break;
			
		}
		if(this._onCmdInvoked != null && this._obj != null){
			this._onCmdInvoked.call(this._obj,cmd.Opcode);

		}
	}	

}